### PR TITLE
Equip messages fix

### LINKED ISSUE #2773: Level check for items returning the wrong message
### Before creating an issue, please ensure:
- [x] This is a bug in the software that resides in this repository, and not a
      support matter (use https://otland.net/forums/support.16/ for support)
- [x] This issue is reproducible without changes to the code in this repository

### Steps to reproduce (include any configuration/script required to reproduce)
1. Try to equip any item that have level requirement while not meeting it

### Expected behaviour
- it should return RETURNVALUE_NOTENOUGHLEVEL
- it should not de-equip the equipped item if the dressing requirements are not met

### Actual behaviour
- it is returning RETURNVALUE_CANNOTBEDRESSED
- it is de-equipping the item even if you can't equip the new one

### PR DESCRIPTION
**Pull request by nekiro**
Fix #2773

### CODE REVIEW

File: src/movement.cpp [hunk 1]
```diff
@@ -659,29 +659,33 @@ uint32_t MoveEvent::RemoveItemField(Item*, Item*, const Position&)
...
-		return 1;
+		return RETURNVALUE_NOERROR;
 	}
 
 	if (!player->hasFlag(PlayerFlag_IgnoreWeaponCheck) && moveEvent->getWieldInfo() != 0) {
+		if (player->getLevel() < moveEvent->getReqLevel()) {
+			return RETURNVALUE_NOTENOUGHLEVEL;
+		}
+
 		if (player->getLevel() < moveEvent->getReqLevel() || player->getMagicLevel() < moveEvent->getReqMagLv()) {
```
Comment (maattch) [inline src/movement.cpp]: duplicate level check?

**Comment by nekiro** [comment]
@Znote are you 100% sure about that? I really tested that and it wasnt deequip my item.

**Comment by Znote** [comment]
leveleq

image

**Comment by nekiro** [comment]
@Znote 
https://nekiro.s-ul.eu/xGISvQIJ.gif

**Comment by danyelvarejao1** [comment]
I have a bug, add an item to the player when it doesn't have a slot to receive it, using player:addItem()
The item will not be sent to the tile.
There will be a userdata for that item, thus leaking "memory".

**Comment by nekiro** [comment]
@danyelvarejao1 you are wrong and how is this even related? If item fails to add to player then it gets deleted. This pr does not change any of that functionality either.
https://github.com/otland/forgottenserver/blob/master/src/luascript.cpp#L9011

**Comment by Znote** [comment]
Still getting weird results. Looks fine on my sorcerer, but not on my knights. 
leveleq2

**Pull request merged**